package server_rpc;

option cc_generic_services = true;

message RegisterRequest {
}

message RegisterResponse {
    required uint32 id = 1;
}

message EchoRequest {
    required uint32 id = 2;
    required string msg = 3;
}

message EchoResponse {
    required string msg = 4;
}

service ServerRpc {
    rpc Register(RegisterRequest) returns (RegisterResponse);
    rpc Echo(EchoRequest) returns (EchoResponse);
}
